src/request.js

```diff
@@ -146,19 +145,47 @@ class Request {
     /**
      * Stores information about an error that occurred during processing of this request.
      *
+     * You should always use Error instances when throwing errors in JavaScript.
+     *
+     * Nevertheless, to improve the debugging experience when using third party libraries
+     * that may not always throw an Error instance, the function performs a type
+     * inspection of the passed argument and attempts to extract as much information
+     * as possible, since just throwing a bad type error makes any debugging rather difficult.
+     *
      * @param {Error|String} errorOrMessage Error object or error message to be stored in the request.
      */
     pushErrorMessage(errorOrMessage) {
-        if (!_.isString(errorOrMessage) && !(errorOrMessage instanceof Error)) {
-            throw new Error('Parameter errorOrMessage must be a String or an instance of Error');
+        let message;
+        const type = typeof errorOrMessage;
+        switch (type) {
```

Comment:
I am just thinking if this is not overkill for these edge cases. The logic could be really simple like:

* if string then use it
* if instance of Error then `obj.message`
* otherwise check if there is `obj.message` and if not use `obj.toString()` which should be always there

Reply:
But maybe why not.

Reply:
I'd agree with Marek, the simpler design the better (and less surprises later). Any reasonable Error implementation should have toString(), and if not, that's the problem of user. BTW are we keeping stack traces for Error objects? IMHO we should, they are crucial for any debugging.